packages/@aws-cdk/aws-ecs/lib/log-drivers/utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Remove undefined values from a dictionary
+ */
+export function removeEmpty<T>(x: { [key: string]: (T | undefined | string) }): { [key: string]: string } {
+    for (const key of Object.keys(x)) {
+      if (!x[key]) {
+        delete x[key];
+      }
+    }
+    return x as any;
+}
+
+/**
+ * Checks that a value is a positive integer
+ */
+export function ensurePositiveInteger(val: number) {
+    if (val && Number.isInteger(val) && val < 0) {
```

Comment:
Wait, does this say what you intend it to say?

If the number is `0.5` now this won't trigger, but you mean to trigger the error, no?

Also please add `!cdk.Token.isUnresolved(...)` checks to every `if`, so that people can use lazy values/cfn parameters if they want to.

I would expect this one to look like:

```ts
if (!cdk.Token.isUnresolved(val) && (!Number.isInteger(val) || val < 0)) {
  throw new Error(...);
}
```

Reply:
Yes as far as i can tell, those properties only support integers. So yes it was intentional, but if you'd like to not be so strict we can allow it to error out at the time of running.

I will add the Token.isUnresolved call.

Reply:
updated for all conditionals in utils.